### PR TITLE
Check for additional file events in watch event poll

### PR DESCRIPTION
**Pull request by vzhukovs**
### What does this PR do?
This changes proposal adds fix for the problem related to sequential handling file watcher events. The problem is in mechanism of receiving events: https://github.com/eclipse/che/blob/master/wsagent/che-core-api-project/src/main/java/org/eclipse/che/api/watcher/server/impl/FileWatcherService.java#L272. We try to get events and continue work with them but we don't check that in this moment we can receive additional events.

Use case to reproduce such problem: use git repository with the following structure:
```
.
├── dir
│   ├── file1
│   ├── file2
│   └── file3
└── other.file
```
clone this repo into che and make outside operation to make the following structure:
```
.
├── dir
└── other.file
```
file watcher detects changes and we call https://github.com/eclipse/che/blob/master/wsagent/che-core-api-project/src/main/java/org/eclipse/che/api/watcher/server/impl/FileWatcherService.java#L272 we receive two events about delete `file1` and `dir`, but in a moment we receive rest two events that is not handled for `file2` and `file3`.

Signed-off-by: Vladyslav Zhukovskyi <vzhukovs@redhat.com>

### What issues does this PR fix or reference?
#11648 

#### Release Notes

N/A

#### Docs PR

N/A

### CODE REVIEW

**Comment by vzhukovskii** [comment]
ci-test

**Comment by vzhukovskii** [comment]
ci-test

**Comment by vzhukovskii** [comment]
ci-build

**Comment by artaleks9** [comment]
Selenium tests execution on Eclipse Che Multiuser on OCP (https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/1408/) doesn't show any regression against this Pull Request.
The bug is not reproduced when checking on local machine.

File: wsagent/che-core-api-project/src/main/java/org/eclipse/che/api/watcher/server/impl/FileWatcherService.java [hunk 1]
```diff
@@ -269,33 +270,35 @@ private void run() {
           continue;
         }
 
-        List<WatchEvent<?>> watchEvents = watchKey.pollEvents();
+        for (List<WatchEvent<?>> watchEvents = watchKey.pollEvents();
```
Comment (vparfonov) [inline wsagent/che-core-api-project/src/main/java/org/eclipse/che/api/watcher/server/impl/FileWatcherService.java]: maybe `while` statement here make code more readable ?

Comment (vzhukovs) [reply wsagent/che-core-api-project/src/main/java/org/eclipse/che/api/watcher/server/impl/FileWatcherService.java]: You mean something like this?:

```
        List<WatchEvent<?>> watchEvents = watchKey.pollEvents();
        while(!watchEvents.isEmpty()) {
          if (suspended.get()) {
            ...
          }

          for (WatchEvent<?> event : watchEvents) {
            ...
          }

          watchEvents = watchKey.pollEvents();
        }
```

**Review by dkulieshov — APPROVED** [review]

**Comment by vzhukovskii** [comment]
ci-build

**Review by vparfonov — APPROVED** [review]

**Pull request merged**